src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/service/FlowDeleteServiceTest.java

```diff
@@ -231,57 +149,48 @@ public void shouldCompleteDeleteOnTimeoutRuleRemoval() {
 
         // 4 times sending 4 rules = 16 requests.
         verify(carrier, times(16)).sendSpeakerRequest(any());
-        verify(flowPathRepository, times(1)).delete(MockitoHamcrest.argThat(
-                Matchers.hasProperty("pathId", is(FORWARD_FLOW_PATH))));
-        verify(flowPathRepository, times(1)).delete(MockitoHamcrest.argThat(
-                Matchers.hasProperty("pathId", is(REVERSE_FLOW_PATH))));
-        verify(flowResourcesManager, times(1)).deallocatePathResources(MockitoHamcrest.argThat(
-                Matchers.hasProperty("forward",
-                        Matchers.<PathResources>hasProperty("pathId", is(FORWARD_FLOW_PATH)))));
-        verify(flowRepository, times(1)).delete(eq(flow));
+        verifyFlowIsMissing(flow);
     }
 
     @Test
     public void shouldFailDeleteOnTimeoutDuringRuleRemoval() {
-        Flow flow = build2SwitchFlow();
-        buildFlowResources();
-
-        FlowDeleteService deleteService = new FlowDeleteService(carrier, persistenceManager,
-                flowResourcesManager, TRANSACTION_RETRIES_LIMIT, SPEAKER_COMMAND_RETRIES_LIMIT);
+        String flowId = makeFlow().getFlowId();
 
-        deleteService.handleRequest("test_key", commandContext, FLOW_ID);
+        FlowDeleteService service = makeService();
 
-        assertEquals(FlowStatus.IN_PROGRESS, flow.getStatus());
-        verify(carrier, times(1)).sendNorthboundResponse(any());
+        service.handleRequest(dummyRequestKey, commandContext, flowId);
+        verifyFlowStatus(flowId, FlowStatus.IN_PROGRESS);
+        verifyNorthboundSuccessResponse(carrier);
 
-        deleteService.handleTimeout("test_key");
+        service.handleTimeout(dummyRequestKey);
 
         verify(carrier, times(4)).sendSpeakerRequest(any());
-        verify(flowResourcesManager, never()).deallocatePathResources(any());
-        verify(flowPathRepository, never()).delete(any());
-        verify(flowRepository, never()).delete(any());
+        // FIXME(surabujin): flow stays in IN_PROGRESS status, any further request can't be handled.
+        //  em... there is no actual handling for timeout event, so FSM will stack in memory forever
```

Comment:
I have traced workflow one more time. The issue is not so serious as mentioned here - speaker worker bolt will produce timeout event's for all unanswered speaker request, and due to this timeout responses delete operation will be completed. So this particular test is excessive because there is no special handling of "global" timeout in delete operation.